### PR TITLE
[New] Set surface navigation constraint

### PR DESCRIPTION
**Pull request by chriswebb09**
## Description

This PR implements **`Set surface navigation constraint `** in `Scenes` category.

`swift/issues/5433`

## How To Test

The sample loads a scene with underground features. Pan and zoom to explore the scene. Observe how the opacity of the base surface is reduced and the navigation constraint is removed, allowing you to pan and zoom through the base surface.

## Screenshots

set-surface-navigation-constraint

### CODE REVIEW

File: Shared/Samples/Set surface navigation constraint/SetSurfaceNavigationConstraintView.swift [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import ArcGIS
+import SwiftUI
+
+struct SetSurfaceNavigationConstraintView: View {
+    /// The Webscene with a portal item.
```
Comment (yo1995) [inline Shared/Samples/Set surface navigation constraint/SetSurfaceNavigationConstraintView.swift:19]:
```suggestion
    /// The web scene with a portal item.
```

File: Shared/Samples/Set surface navigation constraint/SetSurfaceNavigationConstraintView.swift [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    @State private var scene: ArcGIS.Scene = {
+        // Creates the portal item using the id of the Webscene in the ArcGIS portal.
```
Comment (yo1995) [inline Shared/Samples/Set surface navigation constraint/SetSurfaceNavigationConstraintView.swift:21]:
```suggestion
        // Creates the portal item using the ID of the web scene in the ArcGIS portal.
```
When uncertain about a proper noun, use [Esri GIS Dictionary](https://support.esri.com/en-us/gis-dictionary/web-map). Here we don't have an entry for "web scene", but it shares the same concepts as web map.

File: Shared/Samples/Set surface navigation constraint/SetSurfaceNavigationConstraintView.swift [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+        let scene = Scene(item: portalItem)
+        // Sets the navigation constraint on the scene's base surface to unconstrained which allows the camera pass above and below the elevation surface.
+        scene.baseSurface.navigationConstraint = .unconstrained
+        // Sets the opacity so that it is possible to see below the surface.
+        scene.baseSurface.opacity = 0.7
+        return scene
+    }()
+
+    var body: some View {
+        // Displays the Webscene in the SceneView
```
Comment (yo1995) [inline Shared/Samples/Set surface navigation constraint/SetSurfaceNavigationConstraintView.swift:36]:
```suggestion
        // Displays the web scene in the scene view.
```

File: Shared/Samples/Set surface navigation constraint/README.md [hunk 4]
```diff
@@ -0,0 +1,32 @@
+# Set surface navigation constraint
+
+See through terrain in a scene and move the camera underground.
+
+![Screenshot of Set surface navigation constraint sample](set-surface-navigation-constraint.png)
+
+## Use case
+
+By default, a scene's terrain is fully opaque and the camera cannot go underground. To see underground features such as pipes in a utility network, you can lower the opacity of the terrain surface and set the navigation constraint on the surface to allow underground navigation.
```
Comment (CalebRas) [inline Shared/Samples/Set surface navigation constraint/README.md:9]:
```suggestion
By default, a scene's terrain is fully opaque, and the camera cannot go underground. To see underground features such as pipes in a utility network, you can lower the opacity of the terrain surface and set the navigation constraint on the surface to allow underground navigation.
```

File: Shared/Samples/Set surface navigation constraint/README.md [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+
+## How to use the sample
+
+The sample loads a scene with underground features. Pan and zoom to explore the scene. Observe how the opacity of the base surface is reduced and the navigation constraint is removed, allowing you to pan and zoom through the base surface.
+
+## How it works
+
+1. Display an `Scene` in an `SceneView` which contains layers with underground features.
```
Comment (CalebRas) [inline Shared/Samples/Set surface navigation constraint/README.md:17]:
```suggestion
1. Display a `Scene` in a `SceneView`, which contains layers with underground features.
```

File: Shared/Samples/Set surface navigation constraint/README.md [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+3. To allow the camera to go underground, set the surface's navigation constraint to `unconstrained`.
+
+## Relevant API
+
+* Surface
+* Surface.NavigationConstraint
+
+## About the data
+
+This sample shows a [Underground Wellbores and Seismic Events (Point Scene Layer) Webscene](https://www.arcgis.com/home/item.html?id=91a4fafd747a47c7bab7797066cb9272) hosted on ArcGIS Online. The data displays a point scene layer showing underground wellbore paths (green polylines) and seismic events (orange points).
```
Comment (CalebRas) [inline Shared/Samples/Set surface navigation constraint/README.md:28]:
```suggestion
This sample shows an [Underground Wellbores and Seismic Events (Point Scene Layer) Webscene](https://www.arcgis.com/home/item.html?id=91a4fafd747a47c7bab7797066cb9272) hosted on ArcGIS Online. The data displays a point scene layer showing underground wellbore paths (green polylines) and seismic events (orange points).
```

File: Shared/Samples/Set surface navigation constraint/SetSurfaceNavigationConstraintView.swift [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+        let scene = Scene(item: portalItem)
+        // Sets the navigation constraint on the scene's base surface to unconstrained which allows the camera pass above and below the elevation surface.
+        scene.baseSurface.navigationConstraint = .unconstrained
+        // Sets the opacity so that it is possible to see below the surface.
+        scene.baseSurface.opacity = 0.7
+        return scene
+    }()
+
+    var body: some View {
+        // Displays the web scene in the scene view
```
Comment (CalebRas) [inline Shared/Samples/Set surface navigation constraint/SetSurfaceNavigationConstraintView.swift:36]:
```suggestion
        // Displays the web scene in the scene view.
```

File: Samples.xcodeproj/project.pbxproj [hunk 8]
```diff
@@ -183,6 +183,8 @@
 		883C121929C914E100062FF9 /* DownloadPreplannedMapAreaView.swift in Copy Source Code Files */ = {isa = PBXBuildFile; fileRef = E070A0A2286F3B6000F2B606 /* DownloadPreplannedMapAreaView.swift */; };
 		88F93CC129C3D59D0006B28E /* CreateAndEditGeometriesView.swift in Sources */ = {isa = PBXBuildFile; fileRef = 88F93CC029C3D59C0006B28E /* CreateAndEditGeometriesView.swift */; };
 		88F93CC229C4D3480006B28E /* CreateAndEditGeometriesView.swift in Copy Source Code Files */ = {isa = PBXBuildFile; fileRef = 88F93CC029C3D59C0006B28E /* CreateAndEditGeometriesView.swift */; };
+		95F3A52B2C07F09C00885DED /* SetSurfaceNavigationConstraintView.swift in Sources */ = {isa = PBXBuildFile; fileRef = 95F3A52A2C07F09C00885DED /* SetSurfaceNavigationConstraintView.swift */; };
```
Comment (CalebRas) [inline Samples.xcodeproj/project.pbxproj:186]: Can you sort the `Samples` folder in the project by name?

File: Shared/Samples/Set surface navigation constraint/SetSurfaceNavigationConstraintView.swift [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+    /// The web scene with a portal item.
+    @State private var scene: ArcGIS.Scene = {
+        // Creates the portal item using the ID of the web scene in the ArcGIS portal.
+        let portalItem = PortalItem(
+            portal: .arcGISOnline(connection: .anonymous),
+            id: PortalItem.ID("91a4fafd747a47c7bab7797066cb9272")!
+        )
+        // Creates the scene using the portal item.
+        let scene = Scene(item: portalItem)
+        // Sets the navigation constraint on the scene's base surface to unconstrained which allows the camera pass above and below the elevation surface.
```
Comment (CalebRas) [inline Shared/Samples/Set surface navigation constraint/SetSurfaceNavigationConstraintView.swift:28]: This comment is a little long. You might want to break it up into a couple of lines.

File: Shared/Samples/Set surface navigation constraint/SetSurfaceNavigationConstraintView.swift [hunk 10]
```diff
@@ -0,0 +1,43 @@
...
+        scene.baseSurface.navigationConstraint = .unconstrained
+        // Sets the opacity so that it is possible to see below the surface.
+        scene.baseSurface.opacity = 0.7
+        return scene
+    }()
+
```
Comment (CalebRas) [inline Shared/Samples/Set surface navigation constraint/SetSurfaceNavigationConstraintView.swift:34]:
```suggestion
    
```

It looks like this line is missing indentation. You can check for this by selecting the whole file and using Control+I.

**Review by CalebRas — COMMENTED** [review]
Seems to work well on all platforms, just a few housekeeping items:

**Review by CalebRas — APPROVED** [review]

**Review by yo1995 — APPROVED** [review]

**Pull request merged**